src/MatrixClient.ts

```diff
@@ -726,6 +740,21 @@ export class MatrixClient extends EventEmitter {
             .then(response => response["content_uri"]);
     }
 
+    /**
+     * Download content from the homeserver's media repository.
+     * @param {string} mxcUrl the mxcUrl URL.
+     * @param {string} allowRemote Indicates to the server that it should not attempt to fetch the media if it is deemed remote. This is to prevent routing loops where the server contacts itself. Defaults to true if not provided.
+     * @returns {Promise<Buffer>} resolves to a Buffer containing the content
+     */
+    public async downloadContent(mxcUrl: string, allowRemote = true): Promise<{data: Buffer, contentType: string}> {
+        const urlParts = mxcUrl.substr("mxc://".length).split("/");
+        const res = await this.doRequest("GET", `/_matrix/media/r0/download/${urlParts[0]}/${urlParts[1]}`, {allow_remote: allowRemote}, null, null, true, null, true);
```

Comment:
We need to encode the parameters, and handle additional slashes on the media ID.